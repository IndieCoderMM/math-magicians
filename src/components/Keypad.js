import React, { Component } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line
export default class Keypad extends Component {
  render() {
    const { handleClick } = this.props;

    return (
      <>
        <button type="button" onClick={handleClick}>
          AC
        </button>
        <button type="button" onClick={handleClick}>
          +/-
        </button>
        <button type="button" onClick={handleClick}>
          %
        </button>
        <button type="button" onClick={handleClick} className="operator">
          รท
        </button>
        <button type="button" onClick={handleClick}>
          7
        </button>
        <button type="button" onClick={handleClick}>
          8
        </button>
        <button type="button" onClick={handleClick}>
          9
        </button>
        <button type="button" onClick={handleClick} className="operator">
          x
        </button>
        <button type="button" onClick={handleClick}>
          4
        </button>
        <button type="button" onClick={handleClick}>
          5
        </button>
        <button type="button" onClick={handleClick}>
          6
        </button>
        <button type="button" onClick={handleClick} className="operator">
          -
        </button>
        <button type="button" onClick={handleClick}>
          1
        </button>
        <button type="button" onClick={handleClick}>
          2
        </button>
        <button type="button" onClick={handleClick}>
          3
        </button>
        <button type="button" onClick={handleClick} className="operator">
          +
        </button>
        <button type="button" onClick={handleClick} className="wide-btn">
          0
        </button>
        <button type="button" onClick={handleClick}>
          .
        </button>
        <button type="button" onClick={handleClick} className="operator">
          =
        </button>
      </>
    );
  }
}

Keypad.propTypes = { handleClick: PropTypes.func.isRequired };
