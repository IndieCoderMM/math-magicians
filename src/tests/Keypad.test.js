import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Keypad from '../components/Keypad';

describe('Keypad Component', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(
      <Keypad updateCalculation={() => console.log('Calculation updated!')} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should contain 19 buttons', async () => {
    // Arrange
    render(
      <Keypad updateCalculation={() => console.log('UpdateCalculation')} />
    );
    // Act
    const buttons = await screen.findAllByRole('button');
    // Assert
    expect(buttons).toHaveLength(19);
  });
});
