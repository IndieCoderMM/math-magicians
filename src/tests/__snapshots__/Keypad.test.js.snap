// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad Component should render correctly 1`] = `
Array [
  <button
    onClick={[Function]}
    type="button"
  >
    AC
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    %
  </button>,
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    7
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    8
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    9
  </button>,
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    x
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    4
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    5
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    6
  </button>,
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    -
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    1
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    2
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    3
  </button>,
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    +
  </button>,
  <button
    className="wide-btn"
    onClick={[Function]}
    type="button"
  >
    0
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    .
  </button>,
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    =
  </button>,
]
`;
